doc/guide/cert-authentication.xml

```diff
@@ -26,19 +26,19 @@
     <para>Generating the certificates for users is usually done with a certificate management system like
       <ulink url="https://pagure.io/certmonger">certmonger</ulink> or
       <ulink url="https://www.freeipa.org/page/PKI">FreeIPA</ulink>, which are not documented here.
-      For testing purposes, these commands will generate a self-signed certificate/key for the "alice" user:</para>
+      This command generates a simple key and certificate request for the "alice" user:</para>
 
 <programlisting>
-# create self-signed certificate and key
-# some browsers insist on specifying key usage, so it needs a config file
-printf '[req]\ndistinguished_name=dn\nextensions=v3_req\n[dn]\n
-    [v3_req]\nkeyUsage=digitalSignature,keyEncipherment,keyAgreement\n' > /tmp/openssl.cnf
+openssl req -new -newkey rsa:2048 -days 365 -nodes \
+    -keyout alice.key -out alice.csr -subj "/CN=alice"
+</programlisting>
 
-openssl req -x509 -newkey rsa:2048 -days 365 -nodes -keyout alice.key \
-    -out alice.pem -subj "/CN=alice" -config /tmp/openssl.cnf -extensions v3_req
+    <para>Now get this certificate request signed by the Certificate Authority of your Identity
+    Management domain, to get a PEM certificate. Browsers and smart cart utilities accept PKCS#12 format
+    for importing/transfer, so convert the certificate/key pair, and protect it with a transfer
+    password:</para>
 
-# browsers and smart cart utilities accept PKCS#12 format, convert it
-# this needs to set a transfer/import password
+<programlisting>
 openssl pkcs12 -export -password pass:somepassword \
```

Comment:
Can you please do something about this anti-pattern of using a password on the commandline?

Reply:
I added a new commit that drops the `-password`, it will just ask interactively. An admin can figure out how to use `-password file:...` if they need to do this noninteractively. Not pushed yet, let's fist discuss the other comments.

Reply:
Pushed now.